src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const value = String(item[key]);
+
+    if (result[value] === undefined) {
+      result[value] = [];
+    }
+
+    if (result[value]) {
+      result[value].push(item);
+    }
```

Comment:
```suggestion
    if (!result[value]) {
      result[value] = [];
    }

    result[value].push(item);    
  }
```
Навіщо потрібен другий if? Тут дві взаємовиключаючі умови